### PR TITLE
Setup database.yml to be ready to go for Heroku

### PR DESCRIPTION
**Pull request by patbenatar**
Setup everything for Heroku so a fresh app can be deployed successfully.
- rails_12factor replaces rails_stdout_logging
- Add Sendgrid addon and configure its use (remove RecipientInterceptor in staging)
- Default to adding Heroku remotes as ssh-git not http
- Set required ENV vars on Heroku apps
- Update database.yml to work on all environments

### CODE REVIEW

**Comment by patbenatar** [comment]
With a database.yml like this, we no longer need to have it in gitignore

**Comment by patbenatar** [comment]
@taboularasa

**Comment by taboularasa** [comment]
did you confirm this to be working on heroku? if so :shipit:

**Comment by patbenatar** [comment]
@taboularasa damn you called me out on it.. no I hadn't tested it. i will.

**Comment by taboularasa** [comment]
@patbenatar ping

**Comment by patbenatar** [comment]
@taboularasa on it right now!

**Comment by taboularasa** [comment]
@patbenatar okthen

**Comment by patbenatar** [comment]
@taboularasa bye

**Comment by taboularasa** [comment]
@patbenatar byebye

**Comment by patbenatar** [comment]
@taboularasa Ok, this is ready for review again. See updated description above.. this has expanded to address #23 and #24 as well

**Comment by patbenatar** [comment]
@taboularasa Sorry I merged in #36 before getting the squirrel on this..

File: README.md [hunk 1]
```diff
@@ -112,18 +111,18 @@ This:
 * Sets them as `staging` and `production` Git remotes
 * Configures staging with `RACK_ENV` and `RAILS_ENV` environment variables set
   to `staging`
-* Adds the [Rails Stdout Logging][logging-gem] gem
-  to configure the app to log to standard out,
-  which is how [Heroku's logging][heroku-logging] works.
+* Adds the [Rails 12factor][rails-12factor] gem
+* Adds free Sendgrid add-on and configure SMTP settings to use it
```
Comment (taboularasa) [inline README.md]: I've been using mandril on projects lately and found it to be a better product 
compare:
https://addons.heroku.com/sendgrid
https://addons.heroku.com/mandrill

**Comment by taboularasa** [comment]
@patbenatar looks fine, left one comment but :shipit: if you want

File: README.md (hunk 1 above)
Comment (patbenatar) [inline README.md]: I'm sold

**Comment by patbenatar** [comment]
@taboularasa switched to Mandrill. Give it another look please!

**Comment by taboularasa** [comment]
:shipit:

**Pull request merged**